Makefile

```diff
@@ -53,6 +53,7 @@ META_DIR := .make
 
 WHAT ?=
 TESTS ?=
+BRANCH ?=
```

Comment:
I have offered #29320 which backs out the test for undefined variables. which makes some of this moot.